pkg/config/configuration.go

```diff
@@ -369,7 +369,10 @@ func ParseAccessControlSpec(accessControlSpec AccessControlSpec) (*AccessControl
 			Namespace:           appPolicySpec.Namespace,
 			AppOperationActions: operationPolicy,
 		}
-		accessControlList.PolicySpec[aclPolicySpec.AppName] = aclPolicySpec
+
+		// The policy spec can have the same appID which belongs to different namespaces
+		key := aclPolicySpec.AppName + "_" + aclPolicySpec.Namespace
```

Comment:
We use the duplicated code L374 and L496 to generate the key. this could be error-prone in the future. So, I would create new method to generate the key and add the unit-test. L374 and L496 would call this method.

Reply:
fixed